cf-monitord/history.c

```diff
@@ -377,6 +382,7 @@ static Item *NovaReSample(EvalContext *ctx, int slot, const Attributes *attr, co
 
     snprintf(eventname, CF_BUFSIZE - 1, "Sample(%s)", pp->promiser);
     EndMeasure(eventname, start);
+    free((char*)handle);
```

Comment:
No. Don't typecast inside free. The reason why you did it was because your compiler wouldn't let you free it right? You shouldn't free a `const char *`, or typecast it to free it. It is either freed somewhere else, or it's not in heap memory. 

> Avoid using type casts, unless absolutely necessary.
> Usually a compiler warning is better satisfied with correct code rather than using a type cast.

https://github.com/cfengine/core/blob/master/CONTRIBUTING.md#safety--correctness